crates/router/src/core/webhooks/outgoing.rs

```diff
@@ -290,7 +290,7 @@ async fn trigger_webhook_to_merchant(
     );
     logger::debug!(outgoing_webhook_response=?response);
 
-    match delivery_attempt {
+    let updated_event: Option<domain::Event> = match delivery_attempt {
```

Comment:
We're returning `None` from the `match` block only in case of API client errors, after `api_client_error_handler()` is called. And we're returning an error at the end of `api_client_error_handler()`:
https://github.com/juspay/hyperswitch/blob/623cf4c841847f109597ee50017d6cfc0b4d9982/crates/router/src/core/webhooks/outgoing.rs#L736

We should be able to directly return `domain::Event` instead of `Option<domain::Event>` from the match block and this function.

Reply:
Yeah, makes sense.